entrypoint.sh

```diff
@@ -40,4 +40,10 @@ if [ -z "$INPUT_SYNC" ]; then
   CONTAINER_NAME_FLAG="--container"
 fi
 
-az storage blob ${VERB} ${CONNECTION_METHOD} --source ${INPUT_SOURCE_DIR} ${CONTAINER_NAME_FLAG} ${INPUT_CONTAINER_NAME} ${EXTRA_ARGS}
\ No newline at end of file
+az storage blob ${VERB} ${CONNECTION_METHOD} \
+  --source ${INPUT_SOURCE_DIR} ${CONTAINER_NAME_FLAG} ${INPUT_CONTAINER_NAME} ${EXTRA_ARGS} \
+  --output json > upload.json
+
+JSON=`jq -c . upload.json`
```

Comment:
I'm curious why this needs to be piped to a file then passed to `jq`. Can't this all be replaced with the following?

```
BLOBS=$(az storage blob ${VERB} ${CONNECTION_METHOD} --source ${INPUT_SOURCE_DIR} ${CONTAINER_NAME_FLAG} ${INPUT_CONTAINER_NAME} ${EXTRA_ARGS} --output json)

echo "::set-output name=blobs::'$BLOBS'"
```

Why do we need to write the output to a file, then pass to `jq`? Why not write to an environment variable and pass that straight to `::set-output`?

Reply:
On another note... I know that `jq` is provided in the container today. But can we reasonably assume it'll always be available? For example could a future update to mcr.microsoft.com/azure-cli remove `jq` from the container? It appears to be just a "friendly helper to the Azure CLI" more than an actual tool the image provides. I'm not sure we can make the assumption that it'll always be present.

Reply:
I had to use `jq` because I couldn't make `::set-output` work with a multiline json string.
So I used it to compress the json into a single line string.

I agree with your point that it could not be bundled with the azure-cli image. In this case we can check the presence of the `jq` command and install it if needed. I'll send an update.

Reply:
Hmm... Could you just strip all newline characters from the output using `tr`? i.e.

```
JSON=$(az storage blob ... --output json | tr '\n' ' ')
```

That could be more portable than relying on downloading `jq` from the internet at runtime.